experiment/gerrit-onboarder/onboarder.go

```diff
@@ -341,73 +349,131 @@ func ensureProjectConfig(config, host, cur_branch, groupName string) (string, er
 
 }
 
-func updatePojectConfig(host, cur_branch, groupName string) error {
-	data, err := ioutil.ReadFile(projectConfigFile)
+func updatePojectConfig(workDir, host, cur_branch, groupName string) error {
+	data, err := ioutil.ReadFile(path.Join(workDir, projectConfigFile))
 	if err != nil {
-		return fmt.Errorf("faield to read project.config file: %w", err)
+		return fmt.Errorf("failed to read project.config file: %w", err)
 	}
 
-	newData, err := ensureProjectConfig(string(data), host, cur_branch, groupName)
+	newData, err := ensureProjectConfig(workDir, string(data), host, cur_branch, groupName)
 	if err != nil {
 		return fmt.Errorf("failed to ensure updated project config: %w", err)
 	}
-	err = ioutil.WriteFile(projectConfigFile, []byte(newData), 0755)
+	err = ioutil.WriteFile(path.Join(workDir, projectConfigFile), []byte(newData), 0755)
 	if err != nil {
-		return fmt.Errorf("faield to write groups file: %w", err)
+		return fmt.Errorf("failed to write groups file: %w", err)
 	}
 
 	return nil
 }
 
-func fetchMetaConfig(host, repo, branch string) error {
-	if err := bumper.Call(os.Stdout, os.Stderr, "git", "fetch", fmt.Sprintf("sso://%s/%s", host, repo), "refs/meta/config"); err != nil {
+func fetchMetaConfig(host, repo, branch, workDir string) error {
+	if err := execInDir(os.Stdout, os.Stderr, workDir, "git", "fetch", fmt.Sprintf("sso://%s/%s", host, repo), "refs/meta/config"); err != nil {
 		return fmt.Errorf("failed to fetch refs/meta/config, %w", err)
 	}
-	if err := bumper.Call(os.Stdout, os.Stderr, "git", "checkout", "FETCH_HEAD"); err != nil {
+	if err := execInDir(os.Stdout, os.Stderr, workDir, "git", "checkout", "FETCH_HEAD"); err != nil {
 		return fmt.Errorf("failed to checkout FETCH_HEAD, %w", err)
 	}
-	if err := bumper.Call(os.Stdout, os.Stderr, "git", "switch", "-c", branch); err != nil {
+	if err := execInDir(os.Stdout, os.Stderr, workDir, "git", "switch", "-c", branch); err != nil {
 		return fmt.Errorf("failed to switch to new branch, %w", err)
 	}
 
 	return nil
 }
 
+func execInDir(stdout, stderr io.Writer, dir string, cmd string, args ...string) error {
+	(&logrus.Logger{
```

Comment:
If feels a bit weird to initialize this every time. Also, could you explain why setting all these logging configs?

Reply:
Using https://github.com/kubernetes/test-infra/blob/d9d68773cbb3f396035b49b545b8bcefa367155d/prow/cmd/generic-autobumper/bumper/bumper.go#L398 as my reference when creating this function. This feels cleaner than passing the logger around to every function. Should I create the logger as a global var?

Reply:
Ack, nvm then